app/code/Magento/CatalogSearch/Model/Adapter/Mysql/Aggregation/StockSelectProviderInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\CatalogSearch\Model\Adapter\Mysql\Aggregation;
+
+use Magento\Eav\Model\Entity\Attribute\AbstractAttribute;
+use Magento\Framework\DB\Select;
+
+/**
+ * Allow to use different stock query (Service Provider Interface - SPI)
+ */
+interface StockSelectProviderInterface
+{
+    /**
+     * Returns the stock select by current scope
```

Comment:
what is stock select? 
why do we need it? 
Please describe that in comments. Also select is not good extension point, because you don't know fields and tables in select you can reffecrence to